### PR TITLE
Live games : distinguish team spectators and neutral spectators

### PR DESCRIPTION
**Pull request by lucasgautheron**
See issue #48

### CODE REVIEW

File: pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala [hunk 1]
```diff
@@ -75,8 +82,12 @@ case class CompletedServerStateMachine(serverInfoReply: ServerInfoReply, playerI
         frags = frags,
         players = for {
           p <- playerInfoReplies.sortBy(x => (x.flagScore, x.frags)).reverse
-          if p.team.contains(name)
-        } yield CurrentGamePlayer(name = p.name, flags = Option(p.flagScore).filter(_>=0), frags = p.frags)
+          if p.team == name
+        } yield CurrentGamePlayer(name = p.name, flags = Option(p.flagScore).filter(_>=0), frags = p.frags),
+        spectators = Option(for {
+          p <- playerInfoReplies.sortBy(x => (x.flagScore, x.frags)).reverse
+          if (p.team.contains(name) && !activeTeams.contains(p.team))
```
Comment (ScalaWilliam) [inline pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala]: No need for outer brackets for the if --- also you may add another if statement just below as well instead

File: pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala [hunk 2]
```diff
@@ -47,6 +47,13 @@ case class PartialServerStateMachine(serverInfoReplyO: Option[ServerInfoReply] =
 
 case class CompletedServerStateMachine(serverInfoReply: ServerInfoReply, playerInfoReplies: List[PlayerInfoReply], teamInfos: Option[TeamInfos]) extends ServerStateMachine {
   override def next(input: ParsedResponse) = NothingServerStateMachine.next(input)
+  
+  def spectators() : Option[List[String]] = {
```
Comment (ScalaWilliam) [inline pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala]: no need for () - this isn't side-effecting

File: pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala [hunk 3]
```diff
@@ -47,6 +47,13 @@ case class PartialServerStateMachine(serverInfoReplyO: Option[ServerInfoReply] =
...
+      if(teamModes.contains(serverInfoReply.mode))
```
Comment (ScalaWilliam) [inline pinger-client/src/main/scala/acleague/pinger/ServerStateMachine.scala]: Reduce redundancy
```scala
val filteredPlayers = if ( teamModes... ) playerInfoReplies.filter(pi => ...) else pIR.filter(pi => ...)
Option(filteredPlayers.map(_.name)).filter(_.nonEmpty)
```

File: www/render_game.inc.php [hunk 4]
```diff
@@ -35,7 +35,10 @@ function render_game_team($game, $team)
         <ol><?php foreach ($team['players'] as $player) {
                 render_game_team_player($game, $team, $player);
             } ?>
-        </ol>
+            <?php foreach ((isset($team['spectators']) ? $team['spectators'] : []) as $spectator) {
+                render_game_team_player($game, $team, $spectator, true);
+            } ?>
+        </ol
```
Comment (ScalaWilliam) [inline www/render_game.inc.php]: whut

Comment (lucasgautheron) [inline www/render_game.inc.php]: </ol> has a typo this is the problem ? can't see anything else wrong here

Comment (ScalaWilliam) [inline www/render_game.inc.php]: That ol isn't closed properly

Best,
William

Comment (lucasgautheron) [inline www/render_game.inc.php]: This is what i wrote in my comment, but for some reason github ate my html tag.

Comment (ScalaWilliam) [inline www/render_game.inc.php]: Then fix it xD It should be 

```
</ol>
```

We don't want no broken html xD

**Comment by ScalaWilliam** [comment]
LGTM :+1: 

Merge it once you've tested it :D

**Comment by ScalaWilliam** [comment]
:grinning:

**Pull request merged**